### PR TITLE
Remove underscores

### PR DESCRIPTION
**Pull request by pepone**
This PR remove a few remaining underscores 

* Rename encoding fields, Encoding.V1_1 -> Encoding.Version11 
* Replace dispatch methods such iceD_ice_isA -> IceDIceIsA
* Enable `CA1707` for source builds, and disable it for the generated code with a pragma

### CODE REVIEW

**Review by bernardnormier — CHANGES_REQUESTED** [review]
I would rather use V10, V11, V20 for the enumerators. VersionXY sounds too much like an instance of System.Version.

Then, for the IceD methods, this update is not correct because it breaks in the situation where the base and derived interfaces are compiled with different preserve-case metadata. The generated IceD name should not depend on the preserve-case metadata - this is why the existing version uses the operation name as-is from the Slice.

**Comment by pepone** [comment]
> I would rather use V10, V11, V20 for the enumerators. VersionXY sounds too much like an instance of System.Version.

Sounds good.

>Then, for the IceD methods, this update is not correct because it breaks in the situation where the base and derived interfaces are compiled with different preserve-case metadata. The generated IceD name should not depend on the preserve-case metadata - this is why the existing version uses the operation name as-is from the Slice.

I fine with keeping existing convention, but this should not break anything, the call to `operationName(op)` reads the metadata from op an use the appropriate case convention

**Comment by bernardnormier** [comment]
> > Then, for the IceD methods, this update is not correct because it breaks in the situation where the base and derived interfaces are compiled with different preserve-case metadata. The generated IceD name should not depend on the preserve-case metadata - this is why the existing version uses the operation name as-is from the Slice.
> 
> I fine with keeping existing convention, but this should not break anything, the call to `operationName(op)` reads the metadata from op an use the appropriate case convention

We do indeed check the file metadata for each operation, and when the Slice file for a base interface changes, we need to regenerate the code for that interface and all derived interfaces. So your update is correct.

File: cpp/src/slice2cs/Gen.cpp [hunk 1]
```diff
@@ -1080,6 +1080,7 @@ Slice::Gen::Gen(const string& base, const vector<string>& includePaths, const st
     _out << nl << "#pragma warning disable SA1309 // Field names must not begin with underscore";
     _out << nl << "#pragma warning disable SA1312 // Variable names must begin with lower case letter";
     _out << nl << "#pragma warning disable SA1313 // Parameter names must begin with lower case letter";
+    _out << nl << "#pragma warning disable CA1707 // Remove the underscores from member name";
```
Comment (bernardnormier) [inline cpp/src/slice2cs/Gen.cpp:1083]: Why is this necessary?

**Review by bernardnormier — APPROVED** [review]
Looks good.

File: cpp/src/slice2cs/Gen.cpp (hunk 1 above)
Comment (pepone) [reply cpp/src/slice2cs/Gen.cpp:1083]: because `iceP_` param prefix

Comment (pepone) [reply cpp/src/slice2cs/Gen.cpp:1083]: Basically for
```
public static ZeroC.Ice.OutgoingRequestFrame Sleep(ZeroC.Ice.IObjectPrx proxy, int iceP_seconds, global::System.Collections.Generic.IReadOnlyDictionary<string, string>? context) =>
                ZeroC.Ice.OutgoingRequestFrame.WithArgs(
                    proxy,
                    "sleep",
                    idempotent: false,
                    compress: false,
                    format: default,
                    context,
                    iceP_seconds,
                    ZeroC.Ice.OutputStream.IceWriterFromInt);
```

I think should be fine to always use `args` instead of the `iceP_<>` as this is now only for the case we have a single arg.
